### ISSUE TITLE
Unable to add a template value to the Dashboard URL on unified alerts

### ISSUE THREAD

**Issue by togarha** [grafana/grafana#45543]
**What happened**:
I'm trying to add an alert label from a multidimensional unified alerts rule to the DashboardURL alert field, but I'm unable to get it to work.
I tried in several ways but none of them worked for me.

I tried adding the below string to the Dashboard UID:
```
NG-8Ioank/simpledashboard?orgId=1&var-PlatformVar={{ $labels.platform }}
```
![image](https://user-images.githubusercontent.com/24194835/154514790-eb1dfa53-1e72-47ba-87d6-21c64f6beed0.png)

but the `?` character is escaped to its hex value (`%3F`) on the alert, so the address is changed to a wrong parameterized URL:
![image](https://user-images.githubusercontent.com/24194835/154515009-cb713f84-2da9-45f3-ae0c-aa61f2ee16f3.png)

I also tried to use the DashboardURL variable inside the alarm RunBook URL info, so I can use the autogenerated Dashboard URL, but that variable is not expanded on the alert:
```
{{ DashboardURL }}/simpledashboard?orgId=1&var-PlatformVar={{ $labels.platform }}
```
![image](https://user-images.githubusercontent.com/24194835/154515237-3e4156c6-4d63-4957-9d4f-0de8bfed7723.png)

![image](https://user-images.githubusercontent.com/24194835/154515293-4ef7cdef-34ba-4b3b-ab08-f5a257860574.png)

I can set the full URL hardcoded into Runbook URL instead of using the DashboardURL variable but then I have to know the URL where grafana is running on.

**What you expected to happen**:
I expected to be able to add some parameterized info to the DashboardURL, so the information appearing into the alert can point the user to the templated dashboard with the right template values, either using the Dashboard UID field or the RunBookURL field.

**How to reproduce it (as minimally and precisely as possible)**:
Add labels to a multidimensional alert rule Dashboard UID and raise an alert

**Anything else we need to know?**:

**Environment**:
- Grafana version: grafana v8.3.4
- Data source type & version: Microsoft SQL 2019
- OS Grafana is installed on: Red Hat Enterprise Linux release 8.3
- User OS & Browser: Windows10 & google chrome
- Grafana plugins:
- Others:

**Comment by BlackMoon2020** [comment]
Same issue, unable to get label value feed to other field category, description or even custom label.
Datasource influxql.
Grafana: 8.4.2

Yes, my alert is multidimensional.
A. Influxql query about cpu_usage
B. Reduce last
C. $B > 50

Grafana is able to get host, pattern and process_name value and show it under label.
However no matter what formatting I used I am unable to to feed other field.

**Comment by BlackMoon2020** [comment]
My problem above resolved with 8.4.3

**Comment by santihernandezc** [comment]
Hi @togarha, it looks like you are mixing two kinds of template variables: the ones you can use at [alert creation time](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/alerting-rules/alert-annotation-label/#how-are-labels-used), and the ones that you can use in [custom templates](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/message-templating/template-data#template-data/).

You can only use `.DashboardURL` when you define a custom template (plus you're missing a dot at the beginning), but I think you can solve your issue by using the `externalURL` [template function](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/message-templating/template-functions/#template-functions).

In the alert edit page, add an annotation with `{{ externalURL }}simpledashboard?orgId=1&var-PlatformVar={{ $labels.platform }}` as the value.

![Screen Shot 2022-06-09 at 16 41 35](https://user-images.githubusercontent.com/41638679/172933266-a1e099e3-e1da-4a1d-aeb7-07ab10d84e35.png)

When the alert is evaluated, you should get something similar to this in the Grafana UI:

![Screen Shot 2022-06-09 at 16 43 24](https://user-images.githubusercontent.com/41638679/172933856-9a9ec0dc-0718-4c99-ad19-5c8076da5ac4.png)

The alert is then sent to the contact points with this new annotation in the notification body:

![Screen Shot 2022-06-09 at 16 43 42](https://user-images.githubusercontent.com/41638679/172933920-0726c367-e1a9-4467-9f14-e41f1db3fd98.png)

You can also achieve a similar result by creating a custom template and using the `.DashboardURL` field in the `Alert` structure.
